### ISSUE TITLE
Assigning static IP to containers

### ISSUE THREAD

**Issue by djfang** [ansible/ansible-container#350]
Is there a way to assign static IPs to containers? 

The distributed application I deployed needs an IP but since the IP changes when ansible container builds and runs, I therefore need a way to make it static.  I've tried using `links` directive but since it's only one-way mapping, I'm looking for something that's bidirectional. It seems the `networks` directives should solve this but this is currently not supported yet. 

Any thoughts on this will be much appreciated. Cheers and Happy New Year all!

**Comment by chouseknecht** [comment]
@djfang, can you describe your use case a little bit more? Specifically:

- What platform are you deploying to (Swarm, Kubernetes, OpenShift, etc.)?
- What is the expectation for `networks` when you run `ansible-container run --production`?
- What is the expectation for `networks` when you run `shipit`?

**Comment by djfang** [comment]
@chouseknecht Thanks for your reply. Please see my response to your questions below:

### What platform are you deploying to (Swarm, Kubernetes, OpenShift, etc.)?
* I have not tried any of the cloud platforms, I have mainly been testing it locally/on-premise machines.

### What is the expectation for networks when you run ansible-container run --production?

* Assuming my container enviornment looks something like the following:
  * UI container 
  * Database container
  * Data collector container
  * Deployment manager container (manages application configs of the above three containers)
 
* My expectation for networks is as follows assuming network `myNewNetwork` is created:
  * UI container is created using UI’s configuration. It joins the network `myNewNetwork` under the name `ui`.
  * Database container is created using database’s configuration. It joins the network `myNewNetwork` under the name `database`.
  * Data collector container is created using data_collector’s configuration. It joins the network `myNewNetwork` under the name `data_collector`.
  * Deployment management container is created using deployment_manager’s configuration. It joins the network `myNewNetwork` under the name `deployment_manager`.
  * Each container can now look up the hostname ui, database, data_collector, deployment_manager and get back the appropriate container’s IP address. 
    * For example: ui, database, and data_collector containers can connect to deployment_manager container for polling for new configurations using `deployment_manager` hostname AND deployment_manager can connect to `ui`, `database`, and `data_collector` container using their respective hostnames for polling for health, pushing configs, and etc.

### What is the expectation for networks when you run shipit?
* N/A since testing/running locally.

**Comment by chouseknecht** [comment]
@j00bar, thoughts on this?

I'm not sure we want to go down the `networking` road, simply because none of it translates to K8s. Or, maybe we should allow all the network things to function under Docker, and then just ignore them during `shipit`?

**Comment by pheer** [comment]
I was also evaluating ansible containers and looking for network functionality.  In my use case - there is a master and slave nodes that all need to be able to talk to eachother.  Before I've used docker compose containers with 'networks'  (e.g. set each to a common name).  It would be nice to extend this and allow ansible to also provision the container vs simply a dockerfile.  Is there another way I can provision the container using existing ansible roles  (without ansible-container for example).  Mainly I feel it is valuable to have isolated ansible roles that can be run on bare metal machines and/or the same roles be deployed locally for testing.  Hopefully that makes sense...

**Comment by chouseknecht** [comment]
@pheer, for the moment you may be better off using Ansible's [docker modules](https://github.com/ansible/ansible/tree/devel/lib/ansible/modules/cloud/docker).

**Comment by pheer** [comment]
Yep, that is what I'm thinking.  So using those I can setup the cluster and then connect to each container as needed to provision?  You know of a good example showing this use case?   

More info on setup: A vagrant VM with docker/ansible/other tools needed in vm installed via vagrant/ansible provision.  once docker is on vm - 'build' and call 'up' on docker-compose files.  This is where I'm currently at, and now need to further provision these nodes and looking for the best way to accomplish this.
